pkg/remote/aws/vpc_security_group_supplier_test.go

```diff
@@ -4,62 +4,66 @@ import (
 	"context"
 	"testing"
 
+	"github.com/cloudskiff/driftctl/pkg/remote/deserializer"
+
+	"github.com/stretchr/testify/mock"
+
 	awsdeserializer "github.com/cloudskiff/driftctl/pkg/resource/aws/deserializer"
 
+	"github.com/cloudskiff/driftctl/mocks"
 	"github.com/cloudskiff/driftctl/test/goldenfile"
 
 	"github.com/cloudskiff/driftctl/pkg"
 	"github.com/cloudskiff/driftctl/pkg/resource"
 	"github.com/cloudskiff/driftctl/pkg/terraform"
 	"github.com/cloudskiff/driftctl/test"
-	"github.com/cloudskiff/driftctl/test/mocks"
+	mocks2 "github.com/cloudskiff/driftctl/test/mocks"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/ec2"
 )
 
 func TestVPCSecurityGroupSupplier_Resources(t *testing.T) {
 	tests := []struct {
-		test                string
-		dirName             string
-		securityGroupsPages mocks.DescribeSecurityGroupsPagesOutput
-		err                 error
+		test    string
+		dirName string
+		mocks   func(client *mocks.FakeEC2)
+		err     error
 	}{
 		{
 			test:    "no security groups",
 			dirName: "vpc_security_group_empty",
-			securityGroupsPages: mocks.DescribeSecurityGroupsPagesOutput{
-				{
-					true,
-					&ec2.DescribeSecurityGroupsOutput{},
-				},
+			mocks: func(client *mocks.FakeEC2) {
+				client.On("DescribeSecurityGroupsPages",
+					&ec2.DescribeSecurityGroupsInput{},
+					mock.MatchedBy(func(callback func(res *ec2.DescribeSecurityGroupsOutput, lastPage bool) bool) bool {
+						callback(&ec2.DescribeSecurityGroupsOutput{}, true)
+						return true
+					})).Return(nil)
 			},
 			err: nil,
 		},
 		{
 			test:    "with security groups",
 			dirName: "vpc_security_group_multiple",
-			securityGroupsPages: mocks.DescribeSecurityGroupsPagesOutput{
-				{
-					false,
-					&ec2.DescribeSecurityGroupsOutput{
-						SecurityGroups: []*ec2.SecurityGroup{
-							{
-								GroupId: aws.String("sg-0254c038e32f25530"),
+			mocks: func(client *mocks.FakeEC2) {
```

Comment:
👍🏻

Reply:
What about past resource with wrong or different mocks ?

Reply:
Not very important atm, we can do that later in dedicated PR